### PR TITLE
Fix boot_code for ibex compatibility

### PR DESCRIPTION
**Pull request by micprog**

### CODE REVIEW

**Comment by bluewww** [comment]
Running this through gitlabci then I will merge this.

File: boot_code/Makefile [hunk 1]
```diff
@@ -18,7 +18,7 @@
 CC      = riscv32-unknown-elf-gcc
 OBJCOPY = objcopy
 
-CFLAGS   = -Os -g -fno-jump-tables -march=rv32imfcxpulpv2
+CFLAGS   = -Os -g -fno-jump-tables -march=rv32imc -UARCHI_CORE_HAS_PULPV2 -DRV_ISA_RV32
```
Comment (bluewww) [inline boot_code/Makefile:21]: What is `RV_ISA_RV32` for again? Looks like anothe macro name from pulp-sdk/pulp-runtime that is non-descriptive.

File: sim/boot/boot_code.cde [hunk 2]
```diff
@@ -30,64 +30,56 @@
 00000001010000000000000001101111
```
Comment (bluewww) [inline sim/boot/boot_code.cde:30]: I get the same rom, so this is good.

**Review by bluewww — APPROVED** [review]
lgtm, waiting for ci

File: boot_code/Makefile (hunk 1 above)
Comment (micprog) [reply boot_code/Makefile:21]: `RV_32_ISA` is one of the multiple ways to signify not to use custom pulp instructions, i.e. only use rv32imc. I copied these settings from the pulp-runtime modifications, there are a few code sections wrapped with this macro and ifdef there.

**Pull request merged**